.github/workflows/build_wheels_macos_m1.yml

```diff
@@ -0,0 +1,117 @@
+name: Build PYPI wheels for opencv-python on Macos M1
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths-ignore:
+      - '.github/workflows/build_wheels_linux*'
+      - '.github/workflows/build_wheels_windows*'
+      - '.github/workflows/build_wheels_macos.yml'
+  release:
+    types: [published, edited]
+
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macOS-M1]
+        python-version: [3.7, 3.8, 3.9]
+        platform: [x64]
+        with_contrib: [0, 1]
+        without_gui: [0, 1]
+        build_sdist: [0]
+
+    env:
+      SDIST: ${{ matrix.build_sdist || 0 }}
+      ENABLE_HEADLESS: ${{ matrix.without_gui }}
+      ENABLE_CONTRIB: ${{ matrix.with_contrib }}
+      OPENCV_TEST_DATA_PATH: ${{ github.workspace }}/opencv_extra/testdata
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: false
+        fetch-depth: 0
+
+    - name: Update submodules
+      if: github.event_name == 'pull_request'
+      run: git submodule update --remote
+
+    - name: build script
+      run: |
+        git submodule update --init multibuild
```

Comment:
We do not need multibuild here, as we do all the build steps manually. To be honest, I've started thinking we do not need multibuild for all other builders too, as our required steps are simple and we already do them for OSX and windows. But this is a discussion for the other PR)

Reply:
Okay, I will change the flow of the build without multibuild and prepare another PR.